tests/integration/cattletest/core/test_allocation.py

```diff
@@ -67,20 +68,21 @@ def test_spread(admin_client, sim_context, sim_context2, sim_context3):
 
     counts = []
     for i, h in enumerate(hosts):
-        h = admin_client.update(h, {
+        h = internal_test_client.update(h, {
             'computeFree': 10000000
         })
         counts.append(h.computeFree)
 
     containers = []
     for _ in range(len(hosts) * count):
-        c = admin_client.create_container(imageUuid=sim_context['imageUuid'])
+        c = admin_client.create_container(
```

Comment:
In this test, the fundamental thing that you are testing is the change in free compute on a host before and after a container is deployed to it. It doesn't matter which client creates the container. So, for this test, passing in admin_client doesn't add any extra value, but it does add complexity. I'd use *only* the internal_test_client for this test.